mat/format.go

```diff
@@ -87,6 +87,52 @@ func (f formatter) Format(fs fmt.State, c rune) {
 	f.format(f.matrix, f.prefix, f.margin, f.dot, f.squeeze, fs, c)
 }
 
+// formattable holds a matrix and provides methods for formatting values.
+type formattable interface {
+	tabular
+
+	// NewValueFormatter returns a new, zero-valued fmt.Formatter
+	NewValueFormatter(options ...vFormatOption) fmt.Formatter
+
+	// RevalueFormatter sets the value of a formatter to element i, j.
+	// A use is to recycle an existing value formatter for efficiency.
+	RevalueFormatter(f fmt.Formatter, i, j int) fmt.Formatter
+
+	// ValueFormatter returns a new fmt.Formatter for element i, j
+	ValueFormatter(i, j int, options ...vFormatOption) fmt.Formatter
+
+	// Matrix returns the source matrix
+	Mat() tabular
+}
+
+// formattableMatrix is a formattable for floating point-valued matrices, those
+// that satisfy the Matrix interface.
+type formattableMatrix struct{ mat Matrix }
+
+var _ formattable = (*formattableMatrix)(nil)
+
+func (m formattableMatrix) Dims() (r, c int) { return m.mat.Dims() }
+func (m formattableMatrix) Mat() tabular     { return m.mat }
+func (m formattableMatrix) NewValueFormatter(options ...vFormatOption) fmt.Formatter {
+	return formattedFloat(0, options...)
+}
+func (m formattableMatrix) RevalueFormatter(f fmt.Formatter, i, j int) fmt.Formatter {
+	f.(*floatFormatter).value = m.mat.At(i, j)
+	return f
+}
+func (m formattableMatrix) ValueFormatter(i, j int, options ...vFormatOption) fmt.Formatter {
+	return formattedFloat(m.mat.At(i, j), options...)
+}
+
+// tabular type is a two-dimensional array having rows and columns with table
+// elements accessible by a formattable. Details regarding value type and
+// methods for element access are left to the formattable.
+type tabular interface {
+	Dims() (r, c int)
+}
+
+var _ tabular = (Matrix)(nil)
```

Comment:
This is not necessary since `Matrix` is the canonical `interface{ Dim() (r, c int) }` type.